src/main/scala/dibl/proto/TilesConfig.scala

```diff
@@ -83,56 +90,106 @@ import scala.scalajs.js.annotation.{ JSExport, JSExportTopLevel }
+    * @param defaultStitch     tiles.html#footsideStitch, #tileStitch or #headsideStitch
+    */
+  private def setFirstTile(inputMatrix: Seq[String], offsetOfFirstTile: Int, defaultStitch: String): Unit = {
+    for {
+      row <- inputMatrix.indices
+      col <- inputMatrix(row).indices
+    } {
+      val targetCol = col + offsetOfFirstTile
+      val id = Stitches.toID(row, targetCol)
+      val vectorCode = inputMatrix(row)(col)
+      val stitch = if ("-VWXYZ".contains(vectorCode.toUpper)) "-"
+                   else queryFields.getOrElse(id, defaultStitch)
+      if (row < patchHeight && targetCol < totalCols)
+        targetMatrix(row)(targetCol) = Item(
+          id,
+          vectorCode,
+          stitch,
+          row < inputMatrix.length,
+          relativeSources = Matrix.toRelativeSources(vectorCode)
+        )
+    }
+  }
 
-  if (leftMarginWidth > 0) replaceItems(leftMatrix, 0, leftMatrixStitch)
-  if (offsetRightMargin > 0) replaceItems(rightMatrix, offsetRightMargin, rightMatrixStitch)
+  private def copyStitch(targetRow: Int, targetCol: Int, sourceCol: Int, sourceRow: Int): Unit = {
+    targetMatrix(targetRow)(targetCol) = targetMatrix(sourceRow)(sourceCol).copy(isOpaque = false)
+  }
 
-  private def replaceItems(inputMatrix: Seq[String], offset: Int, defaultStitch: String): Unit = {
+  private def repeatSide(offsetOfFirstTile: Int, rows: Int, cols: Int): Unit = {
+    for {col <- offsetOfFirstTile until offsetOfFirstTile + cols} {
+      for {row <- rows until totalRows} {
+        copyStitch(row, col, col, row % rows)
+      }
+    }
+  }
+
+  /**
+    * @param startRow Top position for the new tile within the targetMatrix.
+    * @param startCol Left position for the new tile within the patch.
+    *                 The patch is a subsection of the targetMatrix:
+    *                 from leftMatrixCols until offsetRightMargin.
+    */
+  private def copyCenterTile(startRow: Int, startCol: Int): Unit = {
     for {
-      row <- itemMatrix.indices
-      rSource = row % inputMatrix.length
-      col <- inputMatrix(rSource).indices
+      row <- 0 until centerMatrixRows // row within the tile
+      col <- 0 until centerMatrixCols // col withing the tile
     } {
-      val id = Stitches.toID(rSource, col + offset)
-      val vectorCode = inputMatrix(rSource)(col)
-      val stitch = if ("-VWXYZ".contains(vectorCode.toUpper)) "-"
-                   else fields.getOrElse(id, defaultStitch)
-      itemMatrix(row)(col + offset) = Item(
-        id,
-        vectorCode,
-        stitch,
-        row < inputMatrix.length,
-        relativeSources = Matrix.toRelativeSources(vectorCode)
-      )
+      val targetRow = startRow + row
+      val targetCol = startCol + col + leftMatrixCols
+      val sourceCol = col + leftMatrixCols
+      val sourceRow = row
+      if (stitchIsInsidePatch(targetRow, targetCol)) {
+        copyStitch(targetRow, targetCol, sourceCol, sourceRow)
```

Comment:
```
java.lang.ArrayIndexOutOfBoundsException: 20
	at dibl.proto.TilesConfig.copyStitch(TilesConfig.scala:124)
	at dibl.proto.TilesConfig.$anonfun$copyCenterTile$2(TilesConfig.scala:151)
```
for Whiting pattern [A14](https://d-bl.github.io/GroundForge/tiles?whiting=A14_P87&patchWidth=20&patchHeight=20&tile=5-----5-----,-cvv-b-c-zzb,-x--5-5-5--x,-x-y-5-5-w-x,--5-y-5-w-5-,-e-5-y-zz5-w,01e-y-b-zz21,-x-h-y-y-e-4,--5-w-b-y-5-,-y-w-5-5-y-w,----5-5-5---,-g-y-5-5-w-n,5-----5-----,-cvv-b-c-zzb,-x--5-5-5--x,-x-y-5-5-w-x,--5-y-5-w-5-,-e-5-y-zz5-w,01e-y-b-zz21,-x-h-y-y-e-4,--5-w-b-y-5-,-y-w-5-5-y-w,----5-5-5---,-g-y-5-5-w-n&shiftColsSW=0&shiftRowsSW=24&shiftColsSE=12&shiftRowsSE=12)

Reply:
In copyStitch
targetRow = 8
targetCol = 12
sourceCol = 0
sourceRow = 20

In copyCenterTile
startRow = -12
startCol = 12
row = 20
col = 0
targetRow = 8
targetCol = 12
sourceCol = 0
sourceRow = 20

Reply:
aha, the tile is higher than the patch...

Reply:
Implemented a simple solution: stretch specified patch size to contain at least one tile.